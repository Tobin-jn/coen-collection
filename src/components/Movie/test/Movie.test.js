/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import { Movie, mapStateToProps, mapDispatchToProps } from "../index";

describe("Movie", () => {
  let wrapper;
  let defaultState;

  beforeEach(() => {
    wrapper = shallow(<Movie 
                title={'title'}
                overview={'overview'}
                date={'10-20-2018'}
                poster={'url'} 
              />);

    defaultState = {
      isClicked: false,
    };
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it("should render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(JSON.stringify(wrapper.state())).toEqual(JSON.stringify(defaultState));
  });

  //These two tests aren't passing. Not sure what the reason is....
   
  // it('should call handleCardClick if a Movie is hovered over', () => {
  //   const mockHandleCardClick = jest.fn();

  //   wrapper.instance().handleCardClick = mockHandleCardClick;
    
  //   wrapper.find('.Movie').simulate('mouseenter');

  //   expect(mockHandleCardClick).toHaveBeenCalled();
  // });

  // it('should call handleCardClick if mouse moves off a Movie', () => {
  //   const mockHandleCardClick = jest.fn();

  //   wrapper.instance().handleCardClick = mockHandleCardClick;
    
  //   wrapper.find('.Movie').simulate('mouseleave');

  //   expect(mockHandleCardClick).toHaveBeenCalled();
  // });

  it('should toggle isClicked in state when handleCardClick is called', () => {
    wrapper.setState({isClicked: false});

    wrapper.instance().handleCardClick(); 

    expect(wrapper.state().isClicked).toEqual(true);
  });
});

  describe('mapStateToProps', () => {
    it('should return an object with currentUser', () => {
      const mockState = {
        currentUser: {
          name: 'John',
          id: 11
        }
      }

      const expected = {
        currentUser: {
          name: 'John',
          id: 11
        }
      }

      const mappedToProps = mapStateToProps(mockState)
      expect(mappedToProps).toEqual(expected)
    })
  })









