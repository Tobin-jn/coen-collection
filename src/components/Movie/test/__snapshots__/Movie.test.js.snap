// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    date="10-20-2018"
    overview="overview"
    poster="url"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <i
          className="far fa-star"
          onClick={[Function]}
        />
        <img
          alt="Movie Poster"
          className="movie-poster"
          src="url"
        />
      </div>,
      "className": "Movie",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="far fa-star"
            onClick={[Function]}
          />,
          <img
            alt="Movie Poster"
            className="movie-poster"
            src="url"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "far fa-star",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Movie Poster",
            "className": "movie-poster",
            "src": "url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <i
            className="far fa-star"
            onClick={[Function]}
          />
          <img
            alt="Movie Poster"
            className="movie-poster"
            src="url"
          />
        </div>,
        "className": "Movie",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="far fa-star"
              onClick={[Function]}
            />,
            <img
              alt="Movie Poster"
              className="movie-poster"
              src="url"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-star",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Movie Poster",
              "className": "movie-poster",
              "src": "url",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
